tools/flang2/flang2exe/lldebug.cpp

```diff
@@ -2840,27 +2841,85 @@ lldbg_create_deferred_len_string_type_mdnode(LL_DebugInfo *db, SPTR sptr,
     LL_MDRef type_mdnode =
              lldbg_emit_type(db, DT_INT, sdscsptr, findex, false, false, false);
 
-    /* create a local variable to hold the string length */
-    mdLen = lldbg_create_local_variable_mdnode(
+    if (SCG(sptr) == SC_CMBLK) {
+    /* common blocks are global symbols */
+    /* create a new global variable that can be referred by string length AT
+       of deferred length array.
+       Length of an array is stored in array descriptor
+       (SDSCG(REVMIDLNKG(sptr))) for local scope. A Local deferred array
+       variable has two parts one each for array and its descriptor.
+
+       %mdi_options$p_352 = alloca [1 x i8]*, align 8	==> array
+       %mdi_options$sd_351 = alloca [16 x i64], align 8 ==> descriptor
+
+       But module variables (considered as global variables) are
+       in a single common block variable.
+
+       %struct_modmdi_0_ = type < { [144 x i8]  } >   ==> type of common block
+       @_modmdi_0_ = common global %struct_modmdi_0_  zeroinitializer, align 64,
+                               ==> global variable that holds array and descr
+
+       Here, we find the offset of the descriptor (sdscsptr) in a common block
+       structure and add it with the offset of len member of the
+       descriptor's structure to get the array's length.
+    */
+      strcpy(gbl_arr_len, SYMNAME(sptr));
+      strcat(gbl_arr_len, "_len");
+      SPTR lensptr = getsymbol(gbl_arr_len);
+      /* byte at which length is stored in fortan array  descriptor */
+      const int F90_Desc_byte_len = 8 * (DESC_HDR_BYTE_LEN - DESC_HDR_TAG);
```

Comment:
Is there any predefined constant that could be used instead of `8` here?

Reply:
A few places in this file seem to use this constant. You can try putting that in a common place.

https://github.com/flang-compiler/flang/blob/5a60d344443e38715b7c837de53d9ce2ed78b0d6/tools/flang2/flang2exe/lldebug.cpp#L1342
https://github.com/flang-compiler/flang/blob/5a60d344443e38715b7c837de53d9ce2ed78b0d6/tools/flang2/flang2exe/lldebug.cpp#L1402
https://github.com/flang-compiler/flang/blob/5a60d344443e38715b7c837de53d9ce2ed78b0d6/tools/flang2/flang2exe/lldebug.cpp#L2852

Reply:
Replace the 8 with respective macro on all 4 places. Thanks for the catch.